File: 0198-house-robber/0198-house-robber.swift
class Solution {
    func rob(_ nums: [Int]) -> Int {
        var nums = nums
        
        if nums.count > 2 {
            nums[2] += nums[0]
        }
        
        if nums.count > 3 {
            for i in 3..<nums.count {
                nums[i] += max(nums[i-2], nums[i-3])
            }
        }
        
        return nums.max()!
    }
}

File: 0198-house-robber/0198-house-robber.ts
function rob(nums: number[]): number {
    if (nums.length == 1) {
        return nums[0];
    };

    nums[1] = Math.max(nums[0], nums[1])

    if (nums.length > 2) {
        for (let i = 2; i < nums.length; i++) {
            nums[i] = Math.max((nums[i] + nums[i - 2]), nums[i - 1]);
        };
    };

    return nums[nums.length - 1];
};